### PR TITLE
Claroty xDome Data connector

### PR DESCRIPTION
**Pull request by roysagi**
Required items, please complete
   
   Change(s):
   - Adding new data connector for Claroty xDome

   Reason for Change(s):
   - We want to create a custom CEF connector that will appear in Sentinel's Content Hub.

   Version Updated:
   - NA

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by roysagi** [comment]
> @roysagi please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement
> ### Contribution License Agreement
> This Contribution License Agreement (**“Agreement”**) is agreed to by the party signing below (**“You”**), and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your contributions to Microsoft open source projects. This Agreement is effective as of the latest signature date below.
> 
> 1. **Definitions**.
>    **“Code”** means the computer software code, whether in human-readable or machine-executable form,
>    that is delivered by You to Microsoft under this Agreement.
>    **“Project”** means any of the projects owned or managed by Microsoft and offered under a license
>    approved by the Open Source Initiative ([www.opensource.org](http://www.opensource.org)).
>    **“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any
>    Project, including but not limited to communication on electronic mailing lists, source code control
>    systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
>    discussing and improving that Project, but excluding communication that is conspicuously marked or
>    otherwise designated in writing by You as “Not a Submission.”
>    **“Submission”** means the Code and any other copyrightable material Submitted by You, including any
>    associated comments and documentation.
> 2. **Your Submission**. You must agree to the terms of this Agreement before making a Submission to any
>    Project. This Agreement covers any and all Submissions that You, now or in the future (except as
>    described in Section 4 below), Submit to any Project.
> 3. **Originality of Work**. You represent that each of Your Submissions is entirely Your original work.
>    Should You wish to Submit materials that are not Your original work, You may Submit them separately
>    to the Project if You (a) retain all copyright and license information that was in the materials as You
>    received them, (b) in the description accompanying Your Submission, include the phrase “Submission
>    containing materials of a third party:” followed by the names of the third party and any licenses or other
>    restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
>    guidelines concerning Submissions.
> 4. **Your Employer**. References to “employer” in this Agreement include Your employer or anyone else
>    for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
>    Submission is made in the course of Your work for an employer or Your employer has intellectual
>    property rights in Your Submission by contract or applicable law, You must secure permission from Your
>    employer to make the Submission before signing this Agreement. In that case, the term “You” in this
>    Agreement will refer to You and the employer collectively. If You change employers in the future and
>    desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
>    and secure permission from the new employer before Submitting those Submissions.
> 5. **Licenses**.
> 
> * **Copyright License**. You grant Microsoft, and those who receive the Submission directly or
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
>   Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
>   the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
>   parties.
> * **Patent License**. You grant Microsoft, and those who receive the Submission directly or
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
>   Your patent claims that are necessarily infringed by the Submission or the combination of the
>   Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
>   import or otherwise dispose of the Submission alone or with the Project.
> * **Other Rights Reserved**. Each party reserves all rights not expressly granted in this Agreement.
>   No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
>   granted by implication, exhaustion, estoppel or otherwise.
> 
> 6. **Representations and Warranties**. You represent that You are legally entitled to grant the above
>    licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
>    have disclosed under Section 3). You represent that You have secured permission from Your employer to
>    make the Submission in cases where Your Submission is made in the course of Your work for Your
>    employer or Your employer has intellectual property rights in Your Submission by contract or applicable
>    law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
>    have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
>    You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
>    REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
>    EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
>    PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
>    NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
> 7. **Notice to Microsoft**. You agree to notify Microsoft in writing of any facts or circumstances of which
>    You later become aware that would make Your representations in this Agreement inaccurate in any
>    respect.
> 8. **Information about Submissions**. You agree that contributions to Projects and information about
>    contributions may be maintained indefinitely and disclosed publicly, including Your name and other
>    information that You submit with Your Submission.
> 9. **Governing Law/Jurisdiction**. This Agreement is governed by the laws of the State of Washington, and
>    the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
>    Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
>    exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
>    defenses of lack of personal jurisdiction and forum non-conveniens.
> 10. **Entire Agreement/Assignment**. This Agreement is the entire agreement between the parties, and
>     supersedes any and all prior agreements, understandings or communications, written or oral, between
>     the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

@microsoft-github-policy-service agree company="Claroty"

**Comment by v-prasadboke** [comment]
Hello @roysagi, 

- Solution packaging needs input file and solution metadata which are mandate. 
Input file should be inside Data folder located in Solution folder. 

- You can refer this Solution for more clarification [Solutions/Agari](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Agari)

- Please replace the Data connector file name with Title.

**Comment by roysagi** [comment]
Hi @v-prasadboke - thanks for getting back to me.

A few questions:
1. Is there a way to release this to Sentinel's content hub as standalone content or do we require a Solution here?
2. I Looked at the `SolutionMetadata.json` in the provided example and I am not sure what to put in the `offerId` and if there is a place where I can view the possible `domains` and `verticals` for the `categories`?

Any guidance would be appreciated, Thanks!

**Comment by v-prasadboke** [comment]
> Hi @v-prasadboke - thanks for getting back to me.
> 
> A few questions:
> 
> 1. Is there a way to release this to Sentinel's content hub as standalone content or do we require a Solution here?
> 2. I Looked at the `SolutionMetadata.json` in the provided example and I am not sure what to put in the `offerId` and if there is a place where I can view the possible `domains` and `verticals` for the `categories`?
> 
> Any guidance would be appreciated, Thanks!

You can deploy this Data connector as standalone content. You can move it to Data connector folder in the repo. 
Publisher ID is which your Partner center supports, And offer id can be anything. 
You can go through this Readme for Solution Publishing guidelines. 

As for publishing the Data connector as standalone content is possible as well.

**Comment by v-prasadboke** [comment]
Hello @roysagi, Please confirm once if this Data connector would go as Standalone content or as a Solution

**Comment by roysagi** [comment]
@v-prasadboke - sorry for the delay.
We will proceed with a Solution.
We are having some legal trouble with signing in to the Partner Center in order to open an offer and get the PublisherId.
Can I ping you here once we have everything or would you rather I send a message is some other form?

**Comment by v-prasadboke** [comment]
Hello @roysagi, Thanks for the updates. You can go through this readme file for creating an offer in Partner center 
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/README.md

**Comment by v-prasadboke** [comment]
Hello @roysagi, Commenting if you have any updates on the above conversation.
Thanks

**Comment by v-atulyadav** [comment]
Hi @roysagi,
We're waiting for an update on this. Thanks

**Comment by roysagi** [comment]
@v-atulyadav @v-prasadboke - sorry for missing these messages.
Unfortunately, I am still waiting for partner center access here...

**Comment by v-prasadboke** [comment]
Thanks for the update @roysagi, Please comment back when you have the access

**Comment by v-prasadboke** [comment]
Hello @roysagi, 
Just stopped by to check if you got access to Partner Center

**Comment by v-prasadboke** [comment]
Hello @roysagi, Any updates on the partner center accesss

**Comment by roysagi** [comment]
@v-prasadboke - Yes we have access to the Prtner Center, but legal did not provide me with a commercial account for me to get a publisher id and create an offer - so I pinged them back again.

**Comment by v-prasadboke** [comment]
Ok @roysagi, Thanks for the update.
Until then can you please move this PR to draft state

**Comment by roysagi** [comment]
@v-prasadboke - We have received partner center access and I have added our publisher Id and offer Id - thank you and sorry for the long delay.

**Review by v-prasadboke — DISMISSED** [review]

File: Solutions/Claroty xDome/SolutionMetadata.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+	"publisherId": "claroty",
+	"offerId": "microsoft-sentinel-solution-xdome",
+	"firstPublishDate": "2024-02-01",
+	"providers": ["Claroty"],
+	"categories": {
+		"domains" : ["Security - Threat Intelligence ", "Security - Vulnerability Management"],
```
Comment (v-prasadboke) [inline Solutions/Claroty xDome/SolutionMetadata.json:7]: Hi @roysagi, Please correct domain category

**Review by v-prasadboke — APPROVED** [review]

File: Solutions/Claroty xDome/SolutionMetadata.json (hunk 1 above)
Comment (roysagi) [reply Solutions/Claroty xDome/SolutionMetadata.json:7]: done

Comment (roysagi) [reply Solutions/Claroty xDome/SolutionMetadata.json]: @v-prasadboke - Hi,
It seems I am failing on the domains validation even though the domains I have listed appear [here](https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/sentinel/sentinel-solutions.md?msclkid=9a240b52b11411ec99ae6736bd089c4a#categories-for-microsoft-sentinel-out-of-the-box-content-and-solutions) and I see them used in other solutions such as here:
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/QualysVM/SolutionMetadata.json
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/CTM360/SolutionMetadata.json

**Comment by v-prasadboke** [comment]
Hello @roysagi, 
Unable to pull latest changes to my local. Can you please update the branch from master

**Comment by roysagi** [comment]
> Hello @roysagi, Unable to pull latest changes to my local. Can you please update the branch from master

Hi @v-prasadboke - for some reason I was unable to pull this branch as well, something must have curropted - reopened the PR here: https://github.com/Azure/Azure-Sentinel/pull/10221 on a pulled master as requested

**Pull request closed without merging**